Address I18N-ACTION-90 by adding text from scroll-to-text-fragment#233

aphillips opened the pull request:
This PR borrows and adapts examples by Henri Sivonen. In the course of doing this change, the head matter of the document was revised extensively.

aphillips committed:
Address I18N-ACTION-90 by adding text from scroll-to-text-fragment#233

This PR includes borrowing text from an example by @hsivonen,
which I intend to replace before merging with a better-adapted version.

In addition, some of the text or comments from
WICG/scroll-to-text-fragment#233 are begin adapted into the prose of
this document.

**_Submitting as draft. Not ready for review._**

aphillips committed:
Give credit where it is due.

aphillips committed:
Additional edits

This is a fairly significant refactoring of the introduction
to section 2, plus a retitle of the equivalences section.
I moved the Henri Sivonen example into an example and began
refactoring it (but didn't finish it--it needs to be more illustrative)

More to follow, but committing this for visibility/discussion.

aphillips committed:
(Mostly) finish adapting the example.

aphillips committed:
Major refactor, incl. comments from 2024-09-05 call

aphillips committed:
Clean up example and address comments

aphillips commented:
@hsivonen for visibility. We borrowed your example and adapted it (you are credited in the document). Your comments would be very welcome.

xfq commented on index.html:
```diff
@@ -96,59 +96,111 @@ <h3>Terminology</h3>
         <p class="definition"><dfn data-lt="full text search|full-text search|full text searching">Full-Text Search</dfn> refers to searches that process the entire contents of the textual document or set of documents. Full-text queries perform linguistic searches against text data in full-text indexes by operating on words and phrases based on the rules of a particular language such as English o ...
         <p>Frequently this means that a <a>full-text search</a> employs indexes and natural language processing. When you are using a search engine, you are using a form of full text search. Full text search often breaks natural language text into words or phrases (this is called <a>segmentation</a>) and may apply complex processing to get at the semantic "root" values of words (this is called <a ...
         
+        <p class="definition"><dfn data-lt="natural language processing|NLP">Natural Language Processing</dfn> (<abbr title="natural language processing">NLP</abbr>) refers to the domain of software designed to understand, process, and manipulate human languages (that is, <a>natural language</a>). This is a very wide ranging term. It can cover relatively simple problems, such as word tokenization ...
```
We can consider adding this to the glossary?

Since many aspects of internationalization are related to NLP (tense conversion, pluralization, singularization, parsing the structure of personal names, converting to comparative or superlative, lemmatization and stemming, and so on and so forth), not just string searching, we may also cite this term in other documents.

xfq commented on index.html:
```diff
@@ -96,59 +96,111 @@ <h3>Terminology</h3>
+      <p>There are different types of document searching. One type, called a <a>full text search</a>, is the sort of searching most often found in applications such as a search engine. This type of searching is complex, can be resource intensive, and often depends on processes outside the scope of a given search request.</p> 
     
-      <p>One limited form of full-text search&mdash;and the topic of this document&mdash;is sub-string matching. One familiar form of sub-string matching is the "find" feature of your browser. A sub-string match searches the body ("<a>corpus</a>") of a document with the user's input, seeking a match.</p>
+      <p>A more limited form of text search (and the topic of this document) is <q>sub-string matching</q>. One familiar form of sub-string matching is the <q><em>find</em></q> feature of browsers and other types of user-agent. In browsers, this functionality is often accessed via a key combination such as <kbd>Cmd+F</kbd> or <kbd>Ctrl+F</kbd>. Such a feature might be exposed on the Web via the A ...
```
```suggestion
      <p>A more limited form of text search (and the topic of this document) is <q>sub-string matching</q>. One familiar form of sub-string matching is the <q><em>find</em></q> feature of browsers and other types of user-agent. In desktop browsers, this functionality is often accessed via a key combination such as <kbd>Cmd+F</kbd> or <kbd>Ctrl+F</kbd>. Such a feature might be exposed on the Web via the API <code translate=no>window.find</code>, which is currently not fully standardized, or capabilities such as the proposed scroll-to-text-fragment [[SCROLL-TO-TEXT-FRAGMENT]].</p>
```

aphillips replied:
Thanks. I did slight edits to this in my changes.

xfq commented on index.html:
```diff
@@ -646,6 +698,11 @@ <h4>Visually identical text that is not canonically equivalent</h4>
+      <section id="wordBoundary">
+		  <h3>Word boundaries and "whole word" matching</h3>
+		  
+		  <p>Some languages, such as English or Arabic, use spaces between words. Other languages, such as Chinese, Japanese, or Thai, don't. In many non-spacing languages, computing "whole word" matching depends on the ability to determine word boundaries when the boundaries are not themselves encoded into the text.</p>
```
In orthographies like Vietnamese (Latin) and Lisu (Fraser), spaces separate syllables rather than words. This may also be something that sub-string matching implementations need to consider.

By the way, although Thai doesn't have word separators, spaces are used to separate phrases, so perhaps the term "non-spacing language" is slightly inaccurate. Maybe consider changing the Thai example to Tibetan?

aphillips replied:
Thanks. See changes.

xfq commented on index.html:
```diff
@@ -706,7 +763,9 @@ <h3>Types of Search Option</h3>
-      <p>The W3C Internationalization Working Group and Interest Group, as well as others, provided many comments and suggestions. The Working Group would like to thank: all of the contributors to the Character Model series of documents over the many years of their development. </p>
+      <p>The W3C Internationalization Working Group and Interest Group, as well as others, provided many comments and suggestions. The Working Group would like to thank: all of the contributors to the Character Model series of documents over the many years of their development.</p>
+      
+      <p>The examples in <a href="text-frag-lang">this example</a> were taken from a page authored by Henri Sivonen, as were a number of concepts and ideas recorded by him in <a href="https://github.com/WICG/scroll-to-text-fragment/issues/233">this issue</a>.</p>
```
```suggestion
      <p>The examples in <a href="#text-frag-lang">this example</a> were taken from a page authored by Henri Sivonen, as were a number of concepts and ideas recorded by him in <a href="https://github.com/WICG/scroll-to-text-fragment/issues/233">this issue</a>.</p>
```

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips committed:
Address comments.

aphillips committed:
Improve Henri's example text a bit more
- Convert Turkish examples to a table with code points
- Make the en/de/fi examples use two columns
- Add a hint for the reader
- Add a title to the example

aphillips committed:
Fix egregious typo
